### DISCUSSION TITLE
Welcome to full-blockchain-solidity-course-py Discussions!

### DISCUSSION THREAD

**Discussion by PatrickAlphaC** [smartcontractkit/full-blockchain-solidity-course-py#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by CJ-Rose** [comment]
Hello, whoever is out there : )
I've been learning web development for a few months but I'm really not interested in going any further with old web2 rules. I'm super excited about the infinite potential for blockchain technology and I'm hoping to get the foundational skills to start participating. I'm starting this course today, September 12, and setting the goal to finish it THIS WEEK!  Anybody with me??

**Reply by PatrickAlphaC** [reply]
BOOM!!!

That's such an ambitious goal!! Really looking forward to hearing about the progress you made! Feel free to reach back out here and say how you did!

**Reply by CJ-Rose** [reply]
Lol .. I just spent half a day trying to understand the price conversion in Lesson3, and the second half fighting with my terminal over what's installed or not (Lesson4). I might need a few extra days 😂
Idk if this might be useful for someone else -- I had to use "pip3" for "pip" commands, and "python3" for "python" commands.     ...but only sometimes?? I was able to run "python deploy.py" the first time, when we just printed out the contents of SimpleStorage.sol, but had to use "python3 deploy.py" to compile.

**Reply by PatrickAlphaC** [reply]
oof.... use `python3` and `pip3` for now for everything if you're running into that issue. Maybe I should update this in the issues section? What do you think?

**Reply by CJ-Rose** [reply]
Idk. For sure this speaks more to the issue of me trying to learn how to talk to my computer than the actual course content. It lives in this gap that I keep falling into as a beginner developer - nobody can explain to me why my system is behaving in a particular way, I have to figure it out myself. Which I totally appreciate in theory, it's the only way to really learn. But in practice it's so frustrating bc I lack the fundamental understanding of the system to do the troubleshooting. Same goes for "Read the docs!" - lol I'd read the docs all day long if I could understand what they are saying.
Anyways, I found the pip3 solution pretty easily, and then tried out python3 all by myself. It's not much, but it's rare in this stage that I actually manage to solve my own problem, and I was pretty stoked about it. So idk, maybe leave the low-hanging fruit where it is. 🙃

**Reply by CJ-Rose** [reply]
Hiya 👋
Here I am checking in 3 weeks later....mostly just to say thank you for this huge effort, and especially for your presence and support. Thank you!
As far as my progress.... honestly I'm lost. I made it through about half of the course but I'm feeling altogether overwhelmed. I'm about to go back to the beginning and get a fresh start. I have fallen into a crypto rabbit hole, and in trying to learn about everything that's happening I have lost focus and direction. I have the sense that time spent learning to develop is time lost finding ways to participate in the space, and vice versa.  I'm concerned about investing the time to learn the programming and entering the space as a "blockchain developer" but not understanding what I'm actually supposed to do with the skills, especially as everything seems to be moving so fast! Is it possible to keep up when I feel like I'm already so far behind? Might I be better off just playing Axie Infinity? 🙃
I know this is broad and you're not my life coach, but if you're willing to share any insight here I'd be so grateful.

**Reply by PatrickAlphaC** [reply]
Thanks so much for following up @CJ-Rose!

To be honest, you're doing perfectly so far. There is a _lot_ to learn in this space, and if you can understand all 16 hours perfectly in 3 weeks is like... insane. So don't worry! You're doing exactly the right thing following up on these discussions and asking for help.

> I have the sense that time spent learning to develop is time lost finding ways to participate in the space, and vice versa. 

The opposite of this is actually true. Every second you spend developing, you get a deeper and deeper understanding of the technology you work with. This will give you a leg up even if you decide you don't want to be a developer! Also, as a developer, this is the process everyone goes through to learn and grow.

> but not understanding what I'm actually supposed to do with the skills

You can do whatever you want! You can start a new project, or join a project! People are _dying_ for solidity engineers right now. So after getting a solid footing, and easy starting place is to apply for a job! (if that's what you want to do) Or you can join a hackathon, build fun stuff, it's really whatever you want to do.

> especially as everything seems to be moving so fast! Is it possible to keep up when I feel like I'm already so far behind?
100%. Funny enough, this is _exactly_ how I felt when I first entered the space, and now I'm teaching people how to do it! Things move really fast, but the fundamentals (what we teach in this course) stay about the same. Additionally, there are a ton of areas where people are still exploring the best solutions, and they are moving _much_ slower because of how much thought needs to go into them.

> Might I be better off just playing Axie Infinity? 

That's 100% up to you.

I love these type of questions because that really speaks at the ethos of what we are doing here. A huge part is really dependant on what you want to do, and what you like to do. Maybe a study group would be best for you. Maybe you don't want to be an engineer. Maybe you want to understand engineering so you can speak intelligently as a product manager, or marketer, or designer, or whatever you want to do!

Feel free to start a thread in the discussions section, or join any of the communities linked to the bottom of the `README.md`, and connect with otherwise who are 100% thinking the same things.

In any case, I'm impressed with how far you've gotten! If you get stuck at certain parts, always feel free to ask away here. Again, I'm trying to teach you 2 years of me learning this stuff into 16 hours! So don't feel overwhelmed! Learning is a constant journey. So just chipping away and getting into a solid routine of improving everyday will be helpful even when you finish this course :D

**Reply by CJ-Rose** [reply]
@PatrickAlphaC  WOW 😃  Thank you for this support, honestly I feel so much better right now. Best.Teacher.Ever.
Ok, I'm getting back to work. Thank you!

**Comment by lthibeault** [comment]
I am having at least one issue every Lesson so far that takes hours of troubleshooting and searching the internet for a solution.
I am about to make a Linux virtual machine just to get rid of the windows issues I am having.
I will say I love the discussion section you have, I found one answer and posted my latest issue with Lesson 5 in there. I am really hoping for some help with the solution. 
Thank you for an amazing Video So far, I cant wait to see the Web3.JS Version of this tutorial!

**Reply by PatrickAlphaC** [reply]
Glad to hear you're working so hard to find answers! Please leave an issue, comment, or question with ANYTHING you run into so that others can learn from what you've learnt!

**Comment by nzdog** [comment]
Just want to say thanks Patrick. Course is amazing.

**Reply by PatrickAlphaC** [reply]
Really glad you're enjoying it!

**Comment by MarkusEicher** [comment]
Hello fellow Ethereum and Blockchain enthusiasts. 🤝My name is Markus and I am learning Software Development now concentrating on Web3 and Blockchain. My top interest is Ethereum because I share the vision that this project has. While I'm going through the Tracks at LearnWeb3DAO mainly learning Solidity with JavaScript, TypeScript and Hardhat, I don't want to miss to see the things from the Python and Brownie side. That's why I'm here and I will check out the Hardhat Edition of the Python course later, once it is available. Big thank you to Patrick and all the other people who made this and the JS version possible. 💚 freeCodeCamp is such a gift for all folks learning to code. Nice to meet you all and much fun and success with your learning journey! 🍀

**Reply by cromewar** [reply]
Welcome and We hope you enjoy the course so far.
